src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,125 +1,165 @@
+  invariant(errorHandlerId, 'An errorHandlerId is required');
+  invariant(addonType, 'An add-on type is required.');
+  invariant(authors, 'Authors are required.');
+  invariant(Array.isArray(authors), 'The authors parameter must be an array.');
 
   return {
-    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    type: FETCH_ADDONS_BY_AUTHORS,
     payload: {
       addonType,
       authors,
       errorHandlerId,
-      slug,
+      forAddonSlug,
     },
   };
 };
 
-type LoadOtherAddonsByAuthorsParams = {|
-  slug: string,
+type LoadAddonsByAuthorsParams = {|
   addons: Array<ExternalAddonType>,
+  forAddonSlug?: string,
 |};
 
-type LoadOtherAddonsByAuthorsAction = {|
-  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
-  payload: LoadOtherAddonsByAuthorsParams,
+type LoadAddonsByAuthorsAction = {|
+  type: typeof LOAD_ADDONS_BY_AUTHORS,
+  payload: LoadAddonsByAuthorsParams,
 |};
 
-export const loadOtherAddonsByAuthors = (
-  { addons, slug }: LoadOtherAddonsByAuthorsParams
-): LoadOtherAddonsByAuthorsAction => {
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addons) {
-    throw new Error('A set of add-ons is required.');
-  }
+export const loadAddonsByAuthors = (
+  { addons, forAddonSlug }: LoadAddonsByAuthorsParams
+): LoadAddonsByAuthorsAction => {
+  invariant(addons, 'A set of add-ons is required.');
 
   return {
-    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
-    payload: { slug, addons },
+    type: LOAD_ADDONS_BY_AUTHORS,
+    payload: { addons, forAddonSlug },
   };
 };
 
+export const getAddonsForSlug = (state: State, slug: string) => {
+  const ids = state.byAddonSlug[slug];
+
+  return ids ? ids.map((id) => {
+    return state.byAddonId[id];
+  }) : null;
+};
+
 type Action =
-  | FetchOtherAddonsByAuthorsAction
-  | LoadOtherAddonsByAuthorsAction;
+  | FetchAddonsByAuthorsAction
+  | LoadAddonsByAuthorsAction;
 
 const reducer = (
   state: State = initialState,
   action: Action
 ): State => {
   switch (action.type) {
-    case FETCH_OTHER_ADDONS_BY_AUTHORS:
-      return {
-        ...state,
-        byAddonSlug: {
-          ...state.byAddonSlug,
-          [action.payload.slug]: undefined,
-        },
-      };
-    case LOAD_OTHER_ADDONS_BY_AUTHORS:
-      return {
-        ...state,
-        byAddonSlug: {
-          ...state.byAddonSlug,
-          [action.payload.slug]: action.payload.addons
-            .slice(0, OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE)
-            .map((addon) => createInternalAddon(addon)),
-        },
-      };
+    case FETCH_ADDONS_BY_AUTHORS: {
+      const newState = deepcopy(state);
+
+      if (action.payload.forAddonSlug) {
+        newState.byAddonSlug = {
+          ...newState.byAddonSlug,
+          [action.payload.forAddonSlug]: undefined,
+        };
+      }
+
+      // Reset the data for each author requested.
+      for (const authorUsername of action.payload.authors) {
+        // TODO: Reset the userId here too.
+        newState.byUsername[authorUsername] = [];
```

Comment:
`newState.byUsername[authorUsername]` should be set to `undefined` so you can know the results have *not* been fetched yet. A value of `[]` is an empty result set that was returned from the API; you'd have no way to know if it had been loaded or not.

Reply:
Right on, fair point.